vividus-plugin-web-app-to-rest-api/src/main/java/org/vividus/bdd/steps/integration/ResourceCheckSteps.java

```diff
@@ -122,21 +123,18 @@ private ResourceValidation validate(ResourceValidation r)
     private Stream<ResourceValidation> createResourceValidations(Stream<Element> elements,
             Function<Pair<URI, String>, ResourceValidation> resourceValidationFactory)
     {
-        return elements.map(e ->
-            Pair.of(createUri(getElementAttribute(e, "href").orElseGet(() -> e.attr("src")).trim()), getSelector(e)))
-                       .map(resourceValidationFactory)
-                       .map(rv -> {
-                           URI toCheck = rv.getUri();
-                           boolean validatable = Optional.ofNullable(toCheck.getScheme())
-                                                         .map(s -> ALLOWED_SCHEMES.contains(s))
-                                                         .orElse(false)
-                                                         && !excludeHrefsPattern.matcher(rv.toString()).matches();
-                           if (!validatable)
-                           {
-                               rv.setCheckStatus(CheckStatus.FILTERED);
-                           }
-                           return rv;
-                       });
+        return elements
+                .map(e -> Pair.of(getElementAttribute(e, "href").orElseGet(() -> e.attr("src")).trim(), getSelector(e)))
+                .filter(p -> !p.getKey().isEmpty() || softAssert.recordFailedAssertion(
```

Comment:
Move into separte method
```java
private boolean notEmpty(Pair<String, String>)
{
    return  !p.getKey().isEmpty() || softAssert.recordFailedAssertion("Element by selector " + p.getValue() + " doesn't contain href/src attributes");
}

.filter(this::notEmpty)
```
and use it like:

Reply:
I would not move it, because logic is tightly coupled ('src', 'href' references)